>From 5046c746adbaba3137723a5be38e705f5d29e606 Mon Sep 17 00:00:00 2001
From: Christophe CURIS <christophe.curis@free.fr>
Date: Thu, 31 Dec 2015 12:03:16 +0100
Subject: [PATCH 5/5] Fix incorrect return value compare

In the function obex_client_create_session, when the call to the dbus
function would return a NULL pointer then the program would terminate with
a "Segmentation Fault" message without further explanation, and it is not
sure that in other cases it did behave as expected.

Now the bt-obex command properly display the error message when dbus gives
one and works in the other case.

Signed-off-by: Christophe CURIS <christophe.curis@free.fr>
---
 src/lib/bluez/obex/obex_client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/bluez/obex/obex_client.c b/src/lib/bluez/obex/obex_client.c
index 02b65bb..2a4f20a 100644
--- a/src/lib/bluez/obex/obex_client.c
+++ b/src/lib/bluez/obex/obex_client.c
@@ -144,7 +144,7 @@ const gchar *obex_client_create_session(ObexClient *self, const gchar *destinati
 	g_assert(OBEX_CLIENT_IS(self));
 	const gchar *ret = NULL;
 	GVariant *proxy_ret = g_dbus_proxy_call_sync(self->priv->proxy, "CreateSession", g_variant_new ("(s@a{sv})", destination, args), G_DBUS_CALL_FLAGS_NONE, -1, NULL, error);
-	if (proxy_ret != NULL)
+	if (proxy_ret == NULL)
 		return NULL;
 	proxy_ret = g_variant_get_child_value(proxy_ret, 0);
 	ret = g_variant_get_string(proxy_ret, NULL);
-- 
2.6.4

